Add OpenAQ documentation (#45177)
home-assistant/home-assistant.github.io: source/_integrations/openaq.markdown

---
title: OpenAQ

> description: Instructions on how to integrate OpenAQ air quality data within Home Assistant.

Reviewer: ```suggestion
description: Monitor real-time air quality data from public monitoring stations around the world using OpenAQ.
```

ha_category:
  - Environment
  - Health
  - Sensor
ha_release: 2026.6
ha_iot_class: Cloud Polling
ha_config_flow: true
ha_codeowners:
  - '@jeeftor'
ha_domain: openaq
ha_platforms:
  - sensor
ha_integration_type: service
---

The **OpenAQ** {% term integration %} uses the [OpenAQ](https://openaq.org/) service to monitor recent air quality measurements from public monitoring locations.

OpenAQ collects public air quality data from many providers around the world. In Home Assistant, you can add one or more OpenAQ monitoring locations and create sensors for the latest supported measurements reported by each location.

## Prerequisites

> You need an OpenAQ API key. Sign up for an API key from the [OpenAQ Explorer registration page](https://explore.openaq.org/register). You can manage your API key from your [OpenAQ Explorer account settings](https://explore.openaq.org/account).

Reviewer: ```suggestion
- You need an OpenAQ API key. 
   - Sign up for an API key from the [OpenAQ Explorer registration page](https://explore.openaq.org/register). 
   - You can manage your API key from your [OpenAQ Explorer account settings](https://explore.openaq.org/account).
```

> Treat your API key like a password. Do not share it or publish it in screenshots, logs, or public configuration examples.

Reviewer: ```suggestion
- Treat your API key like a password. 
   - Do not share it or publish it in screenshots, logs, or public configuration examples.
```

{% include integrations/config_flow.md %}

> During setup, enter your OpenAQ API key. Home Assistant validates the key before creating the integration entry.

Reviewer: ```suggestion
{% configuration_basic %}
API key:
  description: Your OpenAQ API key. You can find it in your [OpenAQ Explorer account settings](https://explore.openaq.org/account). Home Assistant validates the key before creating the integration entry.
{% endconfiguration_basic %}
```
We use a configuration basic block for this https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L60

After the integration is added, add at least one monitoring location:

> 1. Under **Settings** > **Devices & services**, select the **OpenAQ** integration.

Reviewer: ```suggestion
1. Under {% my integrations title="**Settings** > **Devices & services**" %}, select the **OpenAQ** integration.
```
We have a [My link](https://developers.home-assistant.io/docs/documenting/general-style-guide#my-links) for this.

2. Select **Add monitoring location**.

> 3. Select a point on the map, set a maximum search radius, and choose one of the suggested monitoring locations. The default radius is 10000 meters, and the maximum radius is 25000 meters.

Reviewer: ```suggestion
3. Select a point on the map, set a maximum search radius, and choose one of the suggested monitoring locations. The default radius is 10000 meters (10 km), and the maximum radius is 25000 meters (25 km).
```

## Supported functionality

The OpenAQ integration creates sensor entities for supported measurements that have a numeric latest value at the selected monitoring location.

Supported sensor measurements:

- Black carbon
- Carbon dioxide
- Carbon monoxide
- Nitrogen dioxide
- Nitrogen monoxide
- Nitrogen oxides
- Ozone
- PM1
- PM2.5
- PM10
- Sulphur dioxide

Home Assistant uses the unit reported by OpenAQ when creating sensor states. Common concentration units, such as parts per million, parts per billion, milligrams per cubic meter, and micrograms per cubic meter, are mapped to Home Assistant units.

Home Assistant also creates a diagnostic distance sensor for each monitoring location. This sensor shows the distance between your Home Assistant location and the OpenAQ monitoring location in your configured length unit, such as miles for US customary units or kilometers for metric units. It is disabled by default.

OpenAQ measurements can include many decimal places. Home Assistant suggests a sensible display precision for these sensors so dashboards stay readable. The raw state value is not rounded by the integration.

## Data updates

> OpenAQ data is polled every 10 minutes.

Reviewer: ```suggestion
OpenAQ data is {% term polling polled %} every 10 minutes.
```
There's a [glossary term reference](https://developers.home-assistant.io/docs/documenting/general-style-guide#glossary-term-reference) for this term :book: .

> Each update fetches the selected monitoring location, the latest measurements for that location, and the sensors available at that location. If Home Assistant cannot reach OpenAQ during an update, the entities for that monitoring location become unavailable until a later update succeeds.

Reviewer: ```suggestion
When the integration loads, Home Assistant fetches the location details, its available sensors, and the latest measurements in parallel. On every subsequent poll, only the latest measurements are fetched. Location and sensor metadata are cached until the integration reloads. If Home Assistant cannot reach OpenAQ during a poll, the entities for that monitoring location become unavailable until the next successful poll.
```

### No monitoring locations found

If the location search does not find a monitoring location, try a larger radius or move the map point closer to a known monitoring station. The maximum search radius is 25000 meters.

You can also add a monitoring location by entering its OpenAQ location ID directly.

### Rate limit exceeded

If Home Assistant reports that the OpenAQ rate limit was exceeded, wait and try again later. If you use the same API key with other tools, those requests also count toward your OpenAQ API usage.

### Expected sensors are missing

Sensors are only created for supported parameters that have a numeric latest value at the selected OpenAQ monitoring location. Check the monitoring location in OpenAQ to confirm that it reports the expected parameter and has recent data.

If OpenAQ starts reporting a supported parameter after the monitoring location was added, reload the OpenAQ integration in Home Assistant.

## Removing the integration

To remove OpenAQ from Home Assistant:

1. Under **Settings** > **Devices & services**, select the **OpenAQ** integration.
2. Select the three-dot menu.

> 3. Select **Delete**.

Reviewer: ```suggestion
This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```
We have a standard reuse element for this.

After the integration is added, add at least one monitoring location:

1. Under {% my integrations title="**Settings** > **Devices & services**" %}, select the **OpenAQ** integration.
2. Select **Add monitoring location**.
3. Select a point on the map, set a maximum search radius, and choose one of the suggested monitoring locations. The default radius is 10000 meters (10 km), and the maximum radius is 25000 meters (25 km).
4. Select the monitoring location to add.

When you search by location, Home Assistant searches outward from the selected map point and shows up to five suggested monitoring locations. The suggestions are ranked by the number of supported sensor measurements first, then by distance. The list shows the supported measurement names and distance for each suggestion. Monitoring locations that do not report any supported measurements are not shown. To add another suggested monitoring location, run **Add monitoring location** again.

You can add multiple OpenAQ monitoring locations to the same OpenAQ integration entry. The same OpenAQ location can only be added once, even across multiple OpenAQ integration entries.

## Supported functionality

> 

Reviewer: ```suggestion
The **OpenAQ** integration provides the following entities.

### Sensors
```
We usually have a section intro sentence here and then a heading per type. https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L93C1-L93C68

## Data updates

OpenAQ data is {% term polling polled %} every 10 minutes.

When the integration loads, Home Assistant fetches the location details, its available sensors, and the latest measurements in parallel. On every subsequent poll, only the latest measurements are fetched. Location and sensor metadata are cached until the integration reloads. If Home Assistant cannot reach OpenAQ during a poll, the entities for that monitoring location become unavailable until the next successful poll.

## Known limitations

OpenAQ does not include every air quality monitoring location or provider in the world. Available measurements depend on the selected monitoring location and the data provided to OpenAQ.

Sensors are created for supported measurements that have a numeric latest value when the monitoring location is set up. If a supported measurement is missing, has no latest value, or reports a non-numeric value, Home Assistant does not create an entity for it. Reload the integration after OpenAQ starts reporting a supported latest value for that location.

Unsupported OpenAQ parameters are ignored.

## Troubleshooting

### Invalid authentication

> If setup reports invalid authentication, confirm that your OpenAQ API key is correct. You can view or rotate your key from your [OpenAQ Explorer account settings](https://explore.openaq.org/account).

Reviewer: ```suggestion
If the setup reports invalid authentication, confirm that your OpenAQ API key is correct. You can view or rotate your key from your [OpenAQ Explorer account settings](https://explore.openaq.org/account).
```
